cpp/FileTransferAgent.cpp

```diff
@@ -250,3 +277,101 @@ void Snowflake::Client::FileTransferAgent::compressSourceFile(
   fclose(sourceFile);
   fclose(destFile);
 }
+
+void Snowflake::Client::FileTransferAgent::download(string *command)
+{
+  m_executionResults = new FileTransferExecutionResult(CommandType::DOWNLOAD,
+    m_largeFilesMeta.size() + m_smallFilesMeta.size());
+
+  std::string createDirIfNotExist = "mkdir -p " +
+    std::string(response.GetLocalLocation());
+  const int err = system(createDirIfNotExist.c_str());
```

Comment:
why you cannot use mkdir function?

Reply:
AFAIK, mkdir system call did not provide "-p", I want to create directories not failing if there is one already existed.

Reply:
I mean this is not portable across platforms.